File: java/src/main/java/br/com/summa/zx/StringBitInputStream.java
/**
 * Zeta-Xi Code - by Einar Saukas
 *
 * See: https://github.com/einar-saukas/Zeta-Xi-Code
 */

package br.com.summa.zx;

public class StringBitInputStream implements BitInputStream {
    private final String input;
    private int index = 0;

    public StringBitInputStream(String input) {
        this.input = input;
    }

    @Override
    public int readBit() {
        return input.charAt(index++) != '0' ? 1 : 0;
    }
}

File: kotlin/src/main/kotlin/br/com/summa/zx/StringBitInputStream.kt
/**
 * Zeta-Xi Code - by Einar Saukas
 *
 * See: https://github.com/einar-saukas/Zeta-Xi-Code
 */

package br.com.summa.zx

class StringBitInputStream(private val input: String) : BitInputStream {
    private var index = 0

    override fun readBit() = if (input[index++] != '0') 1 else 0
}
